src/strategy/flexible-rollout-strategy.ts

```diff
@@ -19,16 +18,14 @@ export class FlexibleRolloutStrategy extends Strategy {
         }
     }
 
-    resolveStickiness(stickiness: string, context: Context): any {
+    resolveStickiness(stickiness: string | undefined, context: Context): any {
         switch (stickiness) {
-            case STICKINESS.userId:
-                return context.userId;
-            case STICKINESS.sessionId:
-                return context.sessionId;
+            case STICKINESS.default:
+                return context.userId || context.sessionId || this.randomGenerator();
             case STICKINESS.random:
                 return this.randomGenerator();
             default:
-                return context.userId || context.sessionId || this.randomGenerator();
+                return stickiness ? resolveContextValue(context, stickiness) : null;
```

Comment:
Would typescript not allow you to do just a `resolveContextValue(context, stickiness)` here? Is there any scenarios where stickiness can be undefined? If that's the case we should probably use the default scenario.

Reply:
You are right. I changed the method (`resolveStickiness`) signature to take a `string` instead of `string | undefined`. This allowed me to make that change.